### DISCUSSION TITLE
pytest development sprint 2024

### DISCUSSION THREAD

**Discussion by The-Compiler** [pytest-dev/pytest#11655] [General] [answered]
# Dedicated sprint repo

**Please see the [separate repository](https://github.com/pytest-dev/sprint/) and [especially the discussions](https://github.com/pytest-dev/sprint/discussions) for up-to-date information on sprint topics.**

**Sign-up is open!** Please [sign up](https://app.formbricks.com/s/clvtdzv8706zdh818vndks441) as soon as possible if you want to join.

Original post for reference below, some information might be outdated!

---

I'm happy to announce that in collaboration with [OMICRON](https://www.omicronenergy.com/en/), I'm aiming to organize another pytest development sprint in 2024!

Back in 2016, we [had an amazing](https://blog.pytest.org/2016/pytest-development-sprint/) development sprint in Freiburg, Germany, with a [successful crowdfunding](https://www.indiegogo.com/projects/python-testing-sprint-mid-2016#/) and 27 participants from five continents meeting up for a week, to hack on pytest 3.0 and tox.

That sprint is where things were implemented which you still likely see today: Renaming the `py.test` binary to `pytest`, the `--setup-only` and `--setup-plan` flags, `-o` to override options — and countless other features, bugfixes, and work on plugins.

In 2024, I'd like to repeat this, with the goal of having a nice mixture of newcomers, plugin authors, people who have worked on pytest itself before, and of course core developers.

We currently are looking at two possible dates:

- **June 17th to 23rd**, 2024
- **June 24th to 30th**, 2024

From what things look like currently, @flub and @asottile would be in, and @Zac-HD as well as @RonnyPfannschmidt answered "maybe".

Given some generous support by OMICRON (one big and two small meeting rooms, reserved for the whole week), and given our [OpenCollective balance of ~$50k USD](https://opencollective.com/pytest), we will be able to host a maximum of 35 people. The details will still need to be discussed, but the idea is that we (partially/fully) **pay for travel and accommodation for participants** from pytest funds, and OMICRON will **offer free lunch** for all participants.

Before we flesh out all the details, I'd like to open up to a bigger audience: **If you might be interested in joining us**, please [fill out this survey](https://nuudel.digitalcourage.de/2tEsEpRcwMNcAXVO) with your preferred date. Obviously nothing of this is binding for now. I'm just trying to figure out how many people we'd be likely need to budget for, before moving into the details.

# Venue

The sprint will be hosted by OMICRON in [Klaus, Vorarlberg, Austria](https://www.openstreetmap.org/#map=15/47.3075/9.6207). Klaus is inhabited by a whopping ~3000 people. Despite being somewhat in the middle of nowhere, it's actually not too hard to reach:

- **~2-2.5h from Zurich Airport**, either:
  - 1h 15min direct Eurocity to Bregenz, then 30min local train from there
  - 10min local train to Zurich main station, 1.5h direct Railjet to Feldkirch, 10min local train from there
- **~1.25h by train from Friedrichshafen**: Train to Lindau, train to Feldkirch, train to Klaus
- **~3h by train from Munich**: 2h direct Eurocity train to Bregenz, 30min local train to Klaus
- **~6.5h by train from Vienna**: just over 6h by direct train (Railjet), and then another 10 minutes in a local train

# Activities

Obviously, we'll hack a lot on pytest! Exact topics will still need to be discussed, but of course we can also sprint on whatever participants  find interesting, as long as it benefits pytest.

We'll make sure to spend time socializing as well obviously. No details here either just yet, but we might have an evening barbecue at OMICRON's venues, and go for a hike or two together while we're there!

**Comment by okken** [comment]
This would be so fun

**Reply by nicoddemus** [reply]
Interested @okken? You would be very welcome I'm sure. As @The-Compiler said, please fill out the survey in case you are interested. 👍

**Reply by okken** [reply]
I actually am. I would love to attend. Either week would probably work, but the earlier one is a titch more convenient. I've added my votes.

**Comment by smitkadvani** [comment]
This would be fun indeed

**Comment by WarrenTheRabbit** [comment]
@The-Compiler  pytest wouldn't get good ROI if you paid for my plane ticket, etc with my current skills and understanding of the codebase, so I definitely won't put my name forward for that. But is there a way to hear conversations and be involved remotely in some way? Would be very interested in that.

**Comment by sospiess** [comment]
Hi @The-Compiler ,
with the Easter school holidays "out of the way", it's only about two months aways. Any updates on the schedule, etc. Just checked and the poll still seems to be open. :)

**Reply by The-Compiler** [reply]
Apologies for the radio silence here... My Python/pytest company trainings somewhat unexpectedly exploded this year, which resulted in me spending a total of 29 nights in March/April in a hotel. With some back-to-back company trainings, I couldn't spend much time at continuing with sprint organization. Additionally, unfortunately nobody else stepped up to help with organizing, so anything not handled by my contact at Omicron will land on my plate it looks like, and said plate is already plenty full... Finally, my contact at Omicron was sick for three weeks and never answered to a mail I sent with some next steps from early March (not blaming them, but it was one more reason this somewhat slipped off my radar the past weeks).

With all that out of the way: I was at Omicron for a training the past two weeks, and thankfully my contact was back at the office this week. We could make some progress, and I plan to set up a sign-up form as soon as I get some time in between PyConDE stuff happening.

The sprint **is happening** (even though maybe in a bit of a smaller scale, consider it a trial run for 2025 maybe?), and it will be **June 17th to 22nd**. More details ASAP.

Greetings from Berlin, and say hi if you're at PyConDE!

**Reply by RonnyPfannschmidt** [reply]
Amazing, thanks for progressing this 

It's great we finally got the dates set.

Take care and some rest

**Reply by Zac-HD** [reply]
Thanks so much for taking on the work of organizing this!

Unfortunately I've had enough stuff come up at work that I won't be able to come this year - I continue to enjoy working at Anthropic, but the hypergrowth doesn't leave much space to relax and I'm already burning a lot of capacity to make it to PyCon US.

Best wishes, and I hope I'll make it to the pytest sprint next year!

**Reply by WarrenTheRabbit** [reply]
I remain interested in being involved in meetings and discussions remotely, if at all possible, even if only in a fly-on-the-wall capacity. No expectation of that though. I'm not a contributor yet, let alone a skilled contributor.

**Reply by The-Compiler** [reply]
@Zac-HD I'd have loved to meet you in person, but it's great that you're taking care of yourself! I really want to repeat this next year, hopefully with things being in place a bit sooner.

@WarrenTheRabbit The location does have pretty decent remote meeting equipment (Webex-based, IIRC), so that should certainly be a possibility. Timezones might be a challenge, though. Given all the thoughts you have around getting more newcomers to the project, I'd love to have you in person as well, but I understand it might be challenging also because it's quite short notice. Maybe if we repeat this next year?

**Reply by WarrenTheRabbit** [reply]
@The-Compiler Remote this year would be fantastic - hopefully I can weather the time zone difference gracefully and learn more about the codebase, roadmap and whatever else. 

Then exploring the option of physically participating next year would be fantastic. 

An update from me is that I've been speaking to social media companies to learn about branding,  community building, sponsorship drives, inclusively and diversity, etc and business coaches that focus on future proofing organisations, etc. My goal is to keep learning about and practising these skills parallel to my career change attempt into software engineering. After a year of working on these skills with communities local to me here in Oz, I should have a lot more ideas and experience relevant to building and supporting the pytest community at next year's sprint.

**Comment by The-Compiler** [comment]
I opened five discussions here with various open topics: https://github.com/pytest-dev/sprint/discussions - I'd love people's inputs on this!

My time unfortunately is still very limited, but hopefully this will lay the foundation of getting a sign-up form ready early next week. I know it's incredibly short-notice especially internationally, but it's the best I could do with everything else getting unexpectedly busy the past months for me. I suppose if nothing else, this will be a "trial run" of some sorts in a somewhat smaller circle, and then we can repeat it better and with more people next year?

**Comment by The-Compiler** [comment, marked as answer]
We just finished the sign-up form, thanks for everyone's patience!

Please [sign up](https://app.formbricks.com/s/clvtdzv8706zdh818vndks441) as soon as possible if you want to join.
